### ISSUE TITLE
X-Original-Content-Length is wrong

### ISSUE THREAD

**Issue by jeffkaufman** [apache/incubator-pagespeed-mod#1380]
```
$ curl www.jefftk.com/tmp.css?PageSpeed=off
body   {   background   :   green }
$ curl -sS www.jefftk.com/tmp.css?PageSpeed=off | wc -c
36
$ curl -sS -D- www.jefftk.com/tmp.css?PageSpeed=off | grep Content-Length
Content-Length: 36
$ curl -sS -D- www.jefftk.com/tmp.css | grep Content-Length
X-Original-Content-Length: 22
Content-Length: 22
```

We're reporting the optimized content length as if it's the original content length.

(I thought this might be load-from-file related, but I'm seeing it both with and without load from file.)

**Comment by jeffkaufman** [comment]
This is in nginx; I haven't tested Apache.

**Comment by hillsp** [comment]
Could this be related to:

https://github.com/pagespeed/mod_pagespeed/commit/5805a9b88b35e4c643a834c4d048efd7236c3222

**Comment by jeffkaufman** [comment]
That could do it!

**Comment by jeffkaufman** [comment]
Wait, no, sorry, that couldn't do it.  It's showing the post-optimization length, not the post-gzip length.

**Comment by jeffkaufman** [comment]
It looks like this was fixed with 1.12, though I don't know which commit fixed it.  Specifically, testing with 1.11.33.4 I see:

```
$ curl -D- -sS  localhost:8050/mod_pagespeed_test/tmp.css
HTTP/1.1 200 OK
Server: nginx/1.11.8
Date: Fri, 20 Jan 2017 20:29:45 GMT
Content-Type: text/css
Content-Length: 36
Last-Modified: Fri, 20 Jan 2017 20:24:07 GMT
Connection: keep-alive
Vary: Accept-Encoding
ETag: "588271e7-24"
Accept-Ranges: bytes

body   {   background   :   green }
$ curl -D- -sS  localhost:8050/mod_pagespeed_test/tmp.css
HTTP/1.1 200 OK
Server: nginx/1.11.8
Content-Type: text/css
Connection: keep-alive
ETag: "588271e7-24"
Accept-Ranges: bytes
Date: Fri, 20 Jan 2017 20:29:45 GMT
Expires: Fri, 20 Jan 2017 20:34:45 GMT
Cache-Control: max-age=300
X-Original-Content-Length: 36
Vary: Accept-Encoding
Vary: Accept-Encoding
Content-Length: 36

body   {   background   :   green }
$ curl -D- -sS  localhost:8050/mod_pagespeed_test/tmp.css
HTTP/1.1 200 OK
Server: nginx/1.11.8
Content-Type: text/css
Connection: keep-alive
Accept-Ranges: bytes
X-Original-Content-Length: 22
Vary: Accept-Encoding
Vary: Accept-Encoding
Content-Length: 22
ETag: W/"PSA-aj-Ybo1X6KozT"
Date: Fri, 20 Jan 2017 20:29:47 GMT
Expires: Fri, 20 Jan 2017 20:34:45 GMT
Cache-Control: max-age=297

body{background:green}
```

And now with 1.12.34.1 I see:

```
$ curl -D- -sS  localhost:8050/mod_pagespeed_test/tmp.css
HTTP/1.1 200 OK
Server: nginx/1.11.8
Date: Fri, 20 Jan 2017 20:31:22 GMT
Content-Type: text/css
Content-Length: 36
Last-Modified: Fri, 20 Jan 2017 20:24:07 GMT
Connection: keep-alive
Vary: Accept-Encoding
ETag: "588271e7-24"
Cache-Control: s-maxage=10
Accept-Ranges: bytes

body   {   background   :   green }
$ curl -D- -sS  localhost:8050/mod_pagespeed_test/tmp.css
HTTP/1.1 200 OK
Server: nginx/1.11.8
Content-Type: text/css
Connection: keep-alive
ETag: "588271e7-24"
Accept-Ranges: bytes
Date: Fri, 20 Jan 2017 20:31:22 GMT
Expires: Fri, 20 Jan 2017 20:36:22 GMT
Vary: Accept-Encoding
Content-Length: 36
Cache-Control: s-maxage=10

body   {   background   :   green }
$ curl -D- -sS  localhost:8050/mod_pagespeed_test/tmp.css
HTTP/1.1 200 OK
Server: nginx/1.11.8
Content-Type: text/css
Connection: keep-alive
Accept-Ranges: bytes
X-Original-Content-Length: 36
Vary: Accept-Encoding
Content-Length: 22
ETag: W/"PSA-aj-Ybo1X6KozT"
Date: Fri, 20 Jan 2017 20:31:24 GMT
Expires: Fri, 20 Jan 2017 20:36:22 GMT
Cache-Control: max-age=297

body{background:green}
```
